src/db.rs

```diff
@@ -120,7 +120,20 @@ impl DbInner {
     ) -> Result<Option<Bytes>, SlateDBError> {
         self.check_error()?;
         let snapshot = self.state.read().snapshot();
-        self.reader.get_with_options(key, options, &snapshot).await
+        self.reader.get_with_options(&key, options, &snapshot).await
+    }
+
+    /// Get the value for a given key.
+    pub async fn multi_get_with_options<K: AsRef<[u8]>>(
+        &self,
+        key: &[K],
+        options: &ReadOptions,
+    ) -> Result<Vec<Option<Bytes>>, SlateDBError> {
```

Comment:
Wondering if it makes sense to return one result per-key. I'm not completely convinced it's a good idea, but it's technically more accurate. Certain keys might fail for various reasons while others might succeed. It does indeed look like the RocksDB [multiget Rust wrapper does this](https://docs.rs/rocksdb/latest/rocksdb/struct.DBCommon.html#method.multi_get). WDYT?